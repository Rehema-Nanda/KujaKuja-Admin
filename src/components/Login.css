
.login-page{
    min-height: calc(100vh - 248px);
}

.login-page h3{
    margin-top: 0px;
    margin-bottom: 16px;
}

.login-page input {
    padding: 16px;
    background-color: #F5F5F5;
    border-radius: 4px;
    border: 0px !important;
    margin-bottom: 16px;
    width: 100%;
}

.login-page button{
    width: 100%;
}

.login-page .login-form{
    background-color: #ffffff;
    border-radius: 8px;
    box-shadow: 0 24px 18px -10px rgba(23,29,33,0.08);
    padding: 24px;
    position: relative;
    top: 50%;
    margin-top: -50%;
}

.login-form .alert {
    margin-top: 16px;
    margin-bottom: 0 !important;
}

.login-row{
    height: calc(100vh - 248px);
}
