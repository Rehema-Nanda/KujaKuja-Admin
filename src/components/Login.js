import React from "react";
import {
    Container, Col, Row, Alert,
} from "reactstrap";

import "./Login.css";
import { StoreContext } from "../StoreContext";

export default class Login extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            errorMsg: "",
            isLoading: false,
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleFormSubmit = this.handleFormSubmit.bind(this);
    }

    handleChange(e) {
        this.setState({
            [e.target.name]: e.target.value,
        });
    }

    async handleFormSubmit(e) {
        this.setState({ isLoading: true });
        const { apiService } = this.context;
        const { username, password } = this.state;

        e.preventDefault();

        try {
            if (!username || !password) {
                throw new Error("Please specify email address and password");
            }

            await apiService.login(username, password);
            this.setState({ isLoading: false });
        }
        catch (error) {
            this.setState({
                errorMsg: error.response ? "Login failed - please try again" : error.message,
                isLoading: false,
            });
        }
    }

    render() {
        const { isLoading } = this.state;
        const { errorMsg } = this.state;

        return (
            <div className="login-page">
                <Container>
                    <Row className="login-row">
                        <Col sm={{ size: 6, offset: 3 }} lg={{ size: 4, offset: 4 }}>
                            <div className="login-form">
                                <h3>Partner Login</h3>

                                <form onSubmit={this.handleFormSubmit}>
                                    <input
                                        placeholder="Username"
                                        name="username"
                                        type="text"
                                        onChange={this.handleChange}
                                    />
                                    <input
                                        placeholder="Password"
                                        name="password"
                                        type="password"
                                        onChange={this.handleChange}
                                    />

                                    <button className="primary" type="submit">
                                        {isLoading ? <div className="loader" /> : "Submit"}
                                    </button>

                                    {errorMsg[0]
                                    && (
                                        <Alert color="danger">
                                            {errorMsg}
                                        </Alert>
                                    )}
                                </form>
                            </div>
                        </Col>
                    </Row>
                </Container>
            </div>
        );
    }
}

Login.contextType = StoreContext;
