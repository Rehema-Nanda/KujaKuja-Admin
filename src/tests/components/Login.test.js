import React from "react";
import { mount } from "enzyme";
import Login from "../../components/Login";

describe("Login component", () => {
    const wrapper = mount(<Login />);

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should handle change on input fields", () => {
        wrapper.find("input").first().simulate(
            "change",
            {
                target: {
                    value: "testUser",
                    name: "username",
                },
            },
        );
        wrapper.find("input").last().simulate(
            "change",
            {
                target: {
                    value: "123",
                    name: "password",
                },
            },
        );

        expect(wrapper.state().username).toBe("testUser");
        expect(wrapper.state().password).toBe("123");
    });

    it.skip("should submit form", () => {
        wrapper.find("form").simulate("submit");
        expect(wrapper.state().loading).toBeTruthy();
    });
});
