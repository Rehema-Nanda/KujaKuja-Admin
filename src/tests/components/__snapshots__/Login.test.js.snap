// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
"<Login>
  <div className=\\"login-page\\">
    <Container tag=\\"div\\">
      <div className=\\"container\\">
        <Row className=\\"login-row\\" tag=\\"div\\">
          <div className=\\"login-row row\\">
            <Col sm={{...}} lg={{...}} tag=\\"div\\" widths={{...}}>
              <div className=\\"col-sm-6 offset-sm-3 col-lg-4 offset-lg-4\\">
                <div className=\\"login-form\\">
                  <h3>
                    Partner Login
                  </h3>
                  <form onSubmit={[Function: bound handleFormSubmit]}>
                    <input placeholder=\\"Username\\" name=\\"username\\" type=\\"text\\" onChange={[Function: bound handleChange]} />
                    <input placeholder=\\"Password\\" name=\\"password\\" type=\\"password\\" onChange={[Function: bound handleChange]} />
                    <button className=\\"primary\\" type=\\"submit\\">
                      Submit
                    </button>
                  </form>
                </div>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </div>
</Login>"
`;
